File: #Task-03/C++/Sub-Task3/SubTask3.cpp
#include <iostream>
using namespace std;
int main(){
        int n;
        cout<<"n: ";
        cin>>n;
        if(n%2!=0){
            int c=n-(n/2)-1;
            int m=1;
            int s1=-1;
            int s2=1;
            for(int g=0;g<n;g++){
                if(c!=0){
                    for(int j=0;j<c;j++){
                        cout<<" ";
                    }
                }
                for (int j=0;j<m;j++) {
                    cout<<"*";
                }
                cout<<endl;
                m+=s2*2;
                c+= s1;
                if(c==0){
                    s1=1;
                }if(m>n){
                    s2=-1;
                    m=n-2;
                }
            }
        }else{
            cout<<"Enter odd number" << endl;
        }
    return 0;
}

File: #Task-03/Rust/Sub-Task3/SubTask3.rs
use std::io;
fn main(){
    println!("Enter n: ");
    let mut input=String::new();
    io::stdin().read_line(&mut input).expect("Failed to read input");
    let n: i32=input.trim().parse().expect("Didnt get number");
    if n%2!=0{
        let mut c=n-(n/2)-1;
        let mut m=1;
        let mut s1=-1;
        let mut s2=1;
        for g in 0..n {
            if c!=0{
                print!("{}"," ".repeat(c as usize));
            }
            println!("{}","*".repeat(m as usize));
            m+=s2*2;
            c+=s1;
            if c==0{
                s1=1;
            }if m>n{
                s2=-1;
                m=n-2;
            }
        }
    } else {
        println!("Enter An Odd Number");
    }
}
